xtask/src/ci/target_matrix.rs

```diff
@@ -32,12 +42,61 @@ pub(crate) fn run(message: String, author: String) -> Result<(), color_eyre::Rep
             std: target.std.map(|b| b as u8),
         })
         .collect::<Vec<_>>();
+
     let json = serde_json::to_string(&matrix)?;
     gha_print(&json);
     gha_output("matrix", &json);
     Ok(())
 }
 
+fn parse_gh_labels(pr: &str) -> cross::Result<Vec<String>> {
+    #[derive(Deserialize)]
+    struct PullRequest {
+        labels: Vec<PullRequestLabels>,
+    }
+
+    #[derive(Deserialize)]
+    struct PullRequestLabels {
+        name: String,
+    }
+    eyre::ensure!(
+        pr.chars().all(|c| c.is_ascii_digit()),
+        "pr should be a number`"
+    );
+    let stdout = Command::new("gh")
+        .args(["pr", "view", pr, "--json", "labels"])
+        .run_and_get_stdout(&mut Verbosity::Quiet.into())?;
+    let pr_info: PullRequest = serde_json::from_str(&stdout)?;
+    Ok(pr_info.labels.into_iter().map(|l| l.name).collect())
+}
+
+fn has_no_ci_target(pr: &str) -> cross::Result<bool> {
+    Ok(parse_gh_labels(pr)?.contains(&"no-ci-targets".to_owned()))
+}
+
+/// Returns the pr(s) associated with this bors commit and the app to use for processing
+fn process_bors_message(message: &str) -> cross::Result<(Vec<&str>, TargetMatrixArgs)> {
+    if let Some(message) = message.strip_prefix("Try #") {
+        let (pr, args) = message
+            .split_once(':')
+            .ok_or_else(|| eyre::eyre!("bors message must start with \"Try #:\""))?;
+        let args = args.trim_start();
+        let app = if !args.is_empty() {
+            TargetMatrixArgs::parse_from(args.split(' '))
+        } else {
+            TargetMatrixArgs::default()
+        };
+        Ok((vec![pr], app))
+    } else if let Some(message) = message.strip_prefix("Merge") {
+        Ok((
+            message.trim().split(" #").skip(1).collect(),
+            TargetMatrixArgs::default(),
```

Comment:
This should still be an issue I believe.

Reply:
yeah, forgot that the message contains more than the title. It has more text :3, need to split on line and then on the contents of that first line

Reply:
I'll look through old PRS to confirm the structure and ensure the tests are run properly.